SerialQueue/SerialQueue/SerialQueue.cs

```diff
@@ -20,10 +21,8 @@ public Task<T> Enqueue<T>(Func<T> function)
         {
             lock (_locker)
             {
-                Task lastTask = null;
-                Task<T> resultTask = null;
-
-                if (_lastTask != null && _lastTask.TryGetTarget(out lastTask))
+                Task<T> resultTask;
+                if (_lastTask.TryGetTarget(out Task lastTask))
```

Comment:
For compatibility with older C# versions let's keep lastTask as separate variable.

Reply:
yes makes sense. Will change this